src/ctap/command.rs

```diff
@@ -134,19 +138,21 @@ impl TryFrom<cbor::Value> for AuthenticatorMakeCredentialParameters {
         let cred_param_vec = read_array(ok_or_missing(param_map.get(&cbor_unsigned!(4)))?)?;
         let mut pub_key_cred_params = vec![];
         for cred_param_map_value in cred_param_vec {
-            let cred_param_map = read_map(cred_param_map_value)?;
-            let cred_type = PublicKeyCredentialType::try_from(ok_or_missing(
-                cred_param_map.get(&cbor_text!("type")),
-            )?)?;
-            let alg = read_integer(ok_or_missing(cred_param_map.get(&cbor_text!("alg")))?)?;
-            pub_key_cred_params.push((cred_type, alg));
+            if let Ok(cred_param) = PublicKeyCredentialParameter::try_from(cred_param_map_value) {
```

Comment:
What's the rationale for ignoring the error code instead of propagating it to the caller? I'm not used to `if let Ok(foo) = ...` pattern, as the error should usually be dealt with.

Reply:
This code is only filtering unknown credential parameters. Maybe the conversion should just return an Option, not a Result then? I wanted to leave it to the application how to deal with different scenarios. If we later support more than one algorithm, it has to be on mod to decide which one to choose. Empty lists could throw an error right away, if you like that better?

Reply:
`PublicKeyCredentialParameter::try_from` will fail if the CBOR value is not a map, or doesn't contain the required fields ("type", "alg"). We usually surface back these errors to the caller. Why should we silently ignore such CBOR values?

The `SignatureAlgorithm::try_from` failing is another class of error. If we don't want to reject unknown algorithms, then maybe `PublicKeyCredentialParameter::try_from` should return `Result<Option<T>, Err>`, where `Ok(None)` is returned in case of unknown algorithm? Or this should be some kind of `RawPublicKeyCredentialParameter` type in which the algorithm is simply encoded as an integer, and then maybe there is a separate conversion function from `RawPublicKeyCredentialParameter` to `Option<PublicKeyCredentialParameter>` which does the validation of the algorithm.

Reply:
Solved by extending SignatureAlgorithm and PublicKeyCredentialType to have an Unknown value.